src/module/apps/sidebar/actor-directory.ts

```diff
@@ -212,21 +212,35 @@ class ActorDirectoryPF2e extends fa.sidebar.tabs.ActorDirectory<ActorPF2e<null>>
     }
 
     protected override async _handleDroppedEntry(target: HTMLElement, data: ActorSidebarDropData): Promise<void> {
-        await super._handleDroppedEntry(target, data);
+        if (!data.uuid) return super._handleDroppedEntry(target, data);
 
-        // Handle dragging members to and from parties (if relevant)
         const toPartyId = htmlClosest(target, "[data-party]")?.dataset.entryId;
+        const toParty = game.actors.get(toPartyId ?? "");
+        const droppedEntry = await this._getDroppedEntryFromData(data);
+
+        // If this is a entry we have to create, and we have a party, then create it here without going upstream
+        if (toParty?.isOfType("party") && !this._entryAlreadyExists(droppedEntry)) {
+            const entry = await ActorPF2e.create(droppedEntry.toObject(), { render: false });
+            if (entry?.isOfType("creature")) {
+                await toParty.addMembers(entry);
+            } else {
+                this.render();
```

Comment:
?

Reply:
To avoid issues due to the render: false above. I can check if droppedEntry is a creature before entering this code instead, but I'll have to cast the newly created actor in the addMembers() function.